contrib/babelfishpg_common/src/collation.c

```diff
@@ -1011,8 +1024,13 @@ get_server_collation_collidx(void)
 {
 	init_server_collation_name();
 	if (NOT_FOUND == server_collation_collidx)
-		server_collation_collidx = find_any_collation(server_collation_name, false);
-
+	{
+		if (database_collation_name)
+			server_collation_collidx = find_any_collation(database_collation_name, false);
```

Comment:
let's not use this notion of server_* for the purpose of database level stuff. Rethink if we can store database_collation_collidx and use it here to return.

Reply:
Also, this helper should be renamed to reflect it can either return database collation or server collation. And update the comment to make it clear what will be returned when.